2019/2019-10-13-brainstorming.md

```diff
@@ -0,0 +1,122 @@
+* We have a number of stakeholders in our products/services:
+  * Competitors: actually using the kit first-hand
+  * Competition team: functionally a middleman, but they give the kit an application/purpose by building a competition around it
+  * Volunteers in general: interacting with the kit comes naturally as part of supporting teams
+* What are the attributes of a kit service that implements this vision?
+  * Easily enabling a minimal viable robot / minimal awesome robot
+  * Kit as a catalyst for students' imagination
+  * Expectations matched to available resources
+  * Kit works out of the box with a team of students (in regard to resources available at school)
+  * Open-ended
+  * Experience themselves
+  * No prescribed process
+  * Should be amazing for students
+  * Should be amazing for spectators
+* What is a subteam?
+  * Group of people collaborating on a piece of work.
+  * No more than about 6 people.
+  * Subteams should have freedom to organise themselves how they see fit.
+  * Committee should oversee operation of the subteam to ensure they are making reasonable progress.
+  * Should keep in touch with the committee on a regular basis, so that they can express their progress and any issues they're having, and we can offer ways to help.
+  * Idea: conferences - each subteam gets a chance to present their work to the rest of the kit team.
+    * Aim to build a sense of community among the whole kit team
+  * Idea: service manuals produced by development teams for use by asset management teams.
+* Ideas for subteams:
+  * Infrastructure team
+    * Provide and maintain managed hosting services to the rest of the organisation
+  * Support team
+    * Provide support to users of the kit, both inside and outside events
+    * Issues trickle down: teams -> mentors/forums -> support team -> domain experts
+  * Asset management/maintenance team
+    * Logistics, storage, tracking, maintenance (to the extent offered by service manuals)
+  * "brain" software team
+    * Robot API, operating system
+  * Development tooling team
+    * IDE, simulator, debuggers (e.g. tablet interface)
+  * Hardware strategy team
+    * Define precisely what is going to be included in the kit
+    * Can construct temporary teams to design and manufacture new boards, working in collaboration with the firmware team.
+  * Firmware team
+    * Maintain and develop the board firmwares.
+  * Robot perception and planning team
+    * Vision and motion planning software
+* Responsibilities of the committee (not an complete list):
+  * Volunteer welfare
+    * We should make it clear to volunteers that we can be contacted as individuals as well as a compound group.
+  * Volunteer onboarding
+    * This could be "outsourced" to a subteam if it's creating too much workload
+* Openness of documentation
+  * Balance needed between "open by default" and social implications of publishing something
+* Kit Team <-> Competition Team challenges
```

Comment:
* Who owns team account management?
  * Who owns forums? (Competition Team)
  * How to implement documentation?
  * How to manage tasks? (https://github.com/srobo/tasks)
  * How to manage communication? (Email, #competition-team?)

Reply:
I think it would be best to discuss these issues in a meeting between the teams, especially as we now know what the kit team will look like.

We will discuss from our end in our meeting on Wednesday.

Reply:
I think it would be best to discuss these issues in a meeting between the teams, especially as we now know what the kit team will look like.

We will discuss from our end in our meeting on Wednesday.